domain/application/errors/errors.go

```diff
@@ -68,4 +72,10 @@ const (
 	// CharmBaseNameNotSupported describes an error that occurs when the charm
 	// base name is not supported.
 	CharmBaseNameNotSupported = errors.ConstError("charm base name not supported")
+
+	// ResourceNotFound describes an error that occurs when a resource is
+	// not found.
+	ResourceNotFound    = errors.ConstError("resource not found")
+	UnknownResourceType = errors.ConstError("unknown resource type resource")
+	ResourceNotValid    = errors.ConstError("resource not valid")
```

Comment:
We should rename RsourceNotValid and call out `ResourceNameNotValid`. I think this one is of particular interest.